### ISSUE TITLE
Feature Request: Auto Keyboard Choosing

### ISSUE THREAD

**Issue by turtlemaster19** [cplaverty/KeitaiWaniKani#42]

**Comment by Mercieral** [comment]
Personally IMO, this seems like a bad idea and the app should be trying to _prevent_ the user from using the iOS Japanese keyboard. The Japanese keyboard makes cheating way too easy because you can type the hiragana and know if you're right/wrong just by seeing if the suggested kanji matches befores submitting...

**Comment by chrisjwelly** [comment]
I agree with @Mercieral . I had experienced practising with the Japanese keyboard on KaniWani and I always noticed I get Kanji suggestions from the keyboard, which is bad for learning. 

Perhaps it would be good if we could have a custom keyboard, where the suggestions are disabled. I'm not sure how feasible this is for iOS though.

**Comment by almercier** [comment]
Agreed. There have been several times where I've felt that using an actual kana keyboard or something like it would be really useful for practice, but I ended up using the suggestions too much which is a big no-no. 

I could see something like a kana "keyboard" on a view that's really just a bunch of buttons injecting text into a field working, but it would be pretty difficult I'd imagine. 

Although this does bring up a good point that @cplaverty is going to have to consider when implementing native lessons/reviews, regarding implementing a native romanji -> kana converter like the one that's currently done by WK in the webview currently.

**Comment by Mercieral** [comment]
Agreed. There have been several times where I've felt that using an actual kana keyboard or something like it would be really useful for practice, but I ended up using the suggestions too much which is a big no-no.

I could see something like a kana "keyboard" on a view that's really just a bunch of buttons injecting text into a field working, but it would be pretty difficult I'd imagine.

Although this does bring up a good point that @cplaverty is going to have to consider when implementing native lessons/reviews, regarding having to implement a native romanji -> kana converter like the one that's done by WK in the webview currently.

**Comment by cplaverty** [comment]
I thought that I might look to port wanakana to swift for the romaji to kana conversion. Or at least start there. 

And I’m not opposed to doing keyboard switching provided the kanji suggestions can be disabled or hidden somehow. Implementing a custom keyboard is certainly an option, but I’ve no idea how much work that would be.

**Comment by almercier** [comment]
I didn't realize WK had the converter JS logic in a public repo. That's pretty awesome.

I'm wondering if some of these settings would hide those suggestions...?
https://stackoverflow.com/a/48324916/11505541

**Comment by Mercieral** [comment]
I didn't realize WK had the converter JS logic in a public repo. That's pretty awesome.

I'm wondering if some of [these settings](https://stackoverflow.com/a/48324916/11505541) would hide those suggestions...?

**Comment by UInt2048** [comment]
> I'm wondering if some of these settings would hide those suggestions...?

Just one is needed - turning off correction (auto-correct)

This can be done programmatically/dynamically, too.
